### PR TITLE
Build gtk.linux.riscv64 native SWT fragment

### PR DESCRIPTION
**Pull request by Sakura286**
With command

```shell
mvn verify -DskipTests=true -Dnative=gtk.linux.riscv64
```

riscv64 artifact can be generated at `binaries/org.eclipse.swt.gtk.linux.riscv64/target/`

### CODE REVIEW

**Comment by Sakura286** [comment]
According to the error log in GitHub CI

```log
Error:  Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.8:package-plugin (default-package-plugin) on project org.eclipse.swt.gtk.linux.riscv64: /home/runner/work/eclipse.platform.swt/eclipse.platform.swt/binaries/org.eclipse.swt.gtk.linux.riscv64/build.properties: bin.includes value(s) [*.so] do not match any files. -> [Help 1]
```

Should the `*.so` files be uploaded manually first?

**Comment by akurtakov** [comment]
*.so files should be generated by the jenkins build. I would leave it to @HannesWell to tell us what is the status here. IIRC equinox is currently blocked by https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4978#note_2820978 and swt has more or less same requirements and should share the same machine.

**Comment by yuzibo** [comment]
Sorry. Can I know when the PR can be merged? maybe this is still blocked by [the ticket](https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4978#note_2820978).:)

**Review by HannesWell — CHANGES_REQUESTED** [review]
For now we need to have the same work-around in the Jenkinsfile of this project like in Equinox to obtain the JDK the natives are built against:
https://github.com/eclipse-equinox/equinox/blob/ef8e0438b0704a5fb8c6cd323081f63c7f04a3ba/Jenkinsfile#L50-L62

After adding this, please also squash all your commits into one with a message covering this change as a whole, e.g.:
`Build gtk.linux.riscv64 native SWT fragment`

**Comment by Sakura286** [comment]
Now the patch is ready for review.

**Review by HannesWell — APPROVED** [review]
I just added the changes from https://github.com/eclipse-platform/eclipse.platform.swt/pull/1490 so that we have a clean version.

Looks good now.

Thank you.

**Comment by HannesWell** [comment]
The initial build cannot succeed in the GH workflows because the native binaries are not yet present.

Submitting for the initial build.

**Pull request merged**
